### DISCUSSION TITLE
integrations/tensorboard/

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#14346] [Docs]
# integrations/tensorboard/

Learn how to integrate YOLOv8 with TensorBoard for real-time visual insights into your model's training metrics, performance graphs, and debugging workflows.

https://docs.ultralytics.com/integrations/tensorboard/

**Comment by lauravchrobak** [comment]
It looks like the Usage code isn't complete. Looks like the first row is cut off?

**Reply by glenn-jocher** [reply]
@lauravchrobak hi there!

Thank you for pointing that out. It seems like the code snippet got truncated. Here's the complete usage example for integrating YOLOv8 with TensorBoard:

```python
from ultralytics import YOLO

# Load a pre-trained model
model = YOLO('yolov8n.pt')

# Train the model
results = model.train(data='coco8.yaml', epochs=100, imgsz=640)
```

After running the training code, you can start TensorBoard to visualize the training metrics:

```bash
tensorboard --logdir path/to/your/tensorboard_logs
```

You can then access the TensorBoard dashboard at [http://localhost:6006/](http://localhost:6006/) to monitor your training progress in real-time.

For more detailed guidance, you can visit our [TensorBoard integration documentation](https://docs.ultralytics.com/integrations/tensorboard/).

If you encounter any issues, please ensure you are using the latest version of the Ultralytics package. If the problem persists, providing a minimum reproducible example would be very helpful. You can find more information on how to create one [here](https://docs.ultralytics.com/help/minimum_reproducible_example).

Happy training! 🚀

**Comment by ruslanraupoff** [comment]
How to forward Tensorboard IP from Docker? Is it possible to use a yolo cli?

**Reply by glenn-jocher** [reply]
@ruslanraupoff to forward TensorBoard from Docker, map the port using `-p 6006:6006` when running your container and start TensorBoard with `tensorboard --logdir /path/to/runs`. Yes, you can use YOLO CLI for training—logs are auto-saved to `runs/` for TensorBoard visualization. See our [Docker Quickstart](https://docs.ultralytics.com/guides/docker-quickstart/) for setup details.

**Comment by HoseinHashemi** [comment]
I found enabling `tensorboard` from within the YOLO training script more comfortable reducing CLI commands.

```
from ultralytics import settings
settings.update({"tensorboard": True})
```

**Reply by glenn-jocher** [reply]
Yes, that’s supported—`settings.update({"tensorboard": True})` is a nice way to enable logging from Python (it persists in your Ultralytics settings across runs), and you can disable it later with `settings.update({"tensorboard": False})`; details on where logs are written and how to view them are in the [TensorBoard integration guide](https://docs.ultralytics.com/integrations/tensorboard/) (feel free to drop any follow-up questions here so the team can see them).
